clinto/parser.py

```diff
@@ -193,9 +195,35 @@ def __init__(self, script_path=None, script_name=None, parsers=None):
                     main_module = module.main.__globals__ if hasattr(module, 'main') else globals()
                     parsers = [v for i, v in chain(six.iteritems(main_module), six.iteritems(vars(module)))
                            if issubclass(type(v), argparse.ArgumentParser)]
+
+            if not parsers:
+                f = tempfile.NamedTemporaryFile()
+                try:
+                    python_code = []
+                    with open(script_path, 'rU') as fi:
+                        for line in fi:
+                            if '.parse_args(' in line:
+                                break
+                            # Crop before parse_args line (prevent script exec/output)
+                            python_code.append(line)
+                    # Write the cropped source file to temporary file
+                    f.write(six.b('\n'.join(python_code)))
+                    f.seek(0)
```

Comment:
I recall temporary files sucking on windows. Anyway to use a cStringIO instead here?